### ISSUE TITLE
Allow nil query values

### ISSUE THREAD

**Issue by vincentjames501** [cemerick/url#14]
First I'd like to say awesome project! We have a project using Stripes and we are trying to parse a referrer header using cemerick. The issue here is that this is a valid url and should be parsed as such.

``` clojure
(url "http://localhost/icmobile/actions/User.action?listAllPageShow=")
=> IllegalArgumentException No value supplied for key: listAllPageShow
```

I think we should adjust the internal query->map function as follows:

``` clojure
(defn- augment 
    [query-vector]
    (if (= 1 (count query-vector))
        [(first query-vector) nil]
        query-vector))

(defn- query->map
  [qstr]
  (when qstr
    (-?>> (.split qstr "&")
      seq
      (mapcat #(.split % "="))
      (map url-decode)
      (augment)
      (apply hash-map))))
```

such that the output is now:

``` clojure
(url "http://localhost/icmobile/actions/User.action?listAllPageShow=abc")
=> #cemerick.url.URL{:protocol "http", :username nil, :password nil, :host "localhost", :port -1, :path "/icmobile/actions/User.action", :query {"listAllPageShow" "abc"}, :anchor nil}
(url "http://localhost/icmobile/actions/User.action?listAllPageShow=")
=> #cemerick.url.URL{:protocol "http", :username nil, :password nil, :host "localhost", :port -1, :path "/icmobile/actions/User.action", :query {"listAllPageShow" nil}, :anchor nil}
```

I'm not a Clojure expert which is why I'm not submitting this as a pull request. If you feel this is acceptable, I'd be happy to submit a pull request.

**Comment by cemerick** [comment]
Looks reasonable. PR welcome.

**Comment by vincentjames501** [comment]
Closing as this looks to be resolved in 0.1.1, I was using an older version.
